### ISSUE TITLE
builds of PR to R_29_Maintenance branch fail

### ISSUE THREAD

**Issue by the-snowwhite** [eclipse-platform/eclipse.platform.swt#885]
**Describe the bug**
builds of PR to R_29_Maintenance branch fails with the error message:

    [ERROR] Failed to execute goal org.eclipse.tycho.extras:tycho-p2-extras-plugin:4.0.2:compare-version-with-baselines
    (compare-attached-artifacts-with-release) on project org.eclipse.swt: Only qualifier changed for 
    (org.eclipse.swt/3.124.100.v20231111-0040). Expected to have bigger x.y.z than what is available in baseline 
    (3.124.100.v20230825-1346)

running the command:

    mvn clean verify --batch-mode -DforkCount=0 -Dcompare-version-with-baselines.skip=false -Dmaven.compiler.failOnWarning=true -Dmaven.test.failure.ignore=true -Dmaven.test.error.ignore=true

**To Reproduce**
A standalone SWT snippet to reproduce the behavior (preferable)
(or) 
Steps to reproduce the behavior.
[Perform a PR to R_29_maintenance branch](https://github.com/eclipse-platform/eclipse.platform.swt/pull/881)

**Expected behavior**
The build succedes

**Environment:**

https://ci.eclipse.org/releng/job/eclipse.platform.swt/view/change-requests/job/PR-881/1/consoleFull
 
**Workaround (or) Additional context**
#858 

Update:
**The purpose of fixing this issue is**

1.being able to enable successful PR builds on any branch
2. Fetching the build artifacts.
3. Closing the PR without a merge.

**Comment by merks** [comment]
You must increase the version in the MANIFEST.MF.   For the maintenance stream it would be add 1 to the third segment for this change.

**Comment by the-snowwhite** [comment]
I can find 2 MANIFEST.MF files in
/home/mib/git/eclipse.platform.swt/bundles/org.eclipse.swt
    
    /home/mib/git/eclipse.platform.swt/bundles/org.eclipse.swt/target/MANIFEST.MF
   /home/mib/git/eclipse.platform.swt/bundles/org.eclipse.swt/META-INF/MANIFEST.MF

only

    /home/mib/git/eclipse.platform.swt/bundles/org.eclipse.swt/target/MANIFEST.MF

contains: 1346
So that must be the correct one.
@merks 
Thanks  Do you know how often this number needs to be increased (like for each new PR) ?

**Comment by merks** [comment]
As you noticed, whatever is in the "target" folder is produced during the build.   The version number needs to be increased if there is a change after the already-released version for the 4.29 stream was built.  The build compares the already-released bundle to the bundle produced by the build.  If they are different, and there is no version change whatsoever, the build fails complaining about only the qualifier changing; the qualifier is computed by the timestamp of the newest commit in the project's source. 

I'm not exactly sure why you are changing the 4.29 stream given the platform doesn't publish/product maintenance releases...

**Comment by the-snowwhite** [comment]
https://github.com/eclipse-platform/eclipse.platform.swt/pull/886#issuecomment-1806923118

**Comment by the-snowwhite** [comment]
https://github.com/eclipse-platform/eclipse.platform.swt/pull/886#issuecomment-1806924491

**Comment by the-snowwhite** [comment]
Seems like this issue is right now blocked by something that not is in the book (Yet).

The purpose of fixing this issue is:

1.being able to enable successful PR builds on any branch
2. Fetching the build artifacts.
3. 3. Closing the PR without a merge. 
@mickaelistria 
Is this correct ?

**Comment by the-snowwhite** [comment]
@merks 
I have opened a New feature request here:
https://github.com/eclipse-platform/eclipse.platform.swt/issues/888

**Comment by the-snowwhite** [comment]
> As you noticed, whatever is in the "target" folder is produced during the build. The version number needs to be increased if there is a change after the already-released version for the 4.29 stream was built. The build compares the already-released bundle to the bundle produced by the build. If they are different, and there is no version change whatsoever, the build fails complaining about only the qualifier changing; the qualifier is computed by the timestamp of the newest commit in the project's source.
> 
> I'm not exactly sure why you are changing the 4.29 stream given the platform doesn't publish/product maintenance releases...

The purpose was not to change the 4.29 stream.
the purpose is being able to Fetch build artifacts from any branch on the repository for testing purposes.
Not merging anything,

**Comment by the-snowwhite** [comment]
> I'm not exactly sure why you are changing the 4.29 stream given the platform doesn't publish/product maintenance releases...
@merks 
Yes I already know that please look at: #858

**Comment by merks** [comment]
Yes, I've seen a great deal of email.

**Comment by mickaelistria** [comment]
THe build is passing if there is no change in it; if you introduce a change, then you need to bump the version as it wasn't done since last 4.29.x release. This is explained in https://wiki.eclipse.org/Version_Numbering
